jenkins/src/cgcloud/jenkins/toil_jenkins_slave.py

```diff
@@ -72,6 +73,14 @@ def __install_parasol( self ):
         sudo( "cp parasol-binaries/* /usr/local/bin" )
         run( "rm -rf parasol-binaries" )
 
+    @fabric_task
+    def __install_yarn ( self ):
+        yarn_version = '2.6.2'
+        run( 'mkdir /opt/apache-hadoop/' )
+        run( 'curl http://apache.osuosl.org/hadoop/common/hadoop-%s/hadoop-%s.tar.gz'
```

Comment:
There is some Apache mirror lookup functionality in SparkBox which might be useful for this. Will need to be pulled up, so it's available here, too. Or put it in an ApacheSoftwareBox mix-in.